plugins/kiro/scripts/hook.sh

```
@@ -0,0 +1,257 @@
+#   3. $HOME/.rogue-env          (per-user / installer-written)
+
+EVENT_ARG="${1:-}"
+SURFACE_ARG="${2:-}"
+
+# Self-locate the plugin root from $0 (the installer writes the absolute path
+# <root>/scripts/hook.sh into the hook file). Fall back to the env token.
+PLUGIN_ROOT="$(CDPATH= cd -- "$(dirname -- "$0")/.." 2>/dev/null && pwd)"
+[ -n "$PLUGIN_ROOT" ] || PLUGIN_ROOT="${KIRO_PLUGIN_ROOT:-${PLUGIN_ROOT:-.}}"
+
+# Env precedence (later wins): bundled → MDM → per-user.
+[ -r "${PLUGIN_ROOT}/env" ] && . "${PLUGIN_ROOT}/env"
```

Shellcheck (0.11.0)

[info] Not following: ./env was not specified as input (see shellcheck -x).

(SC1091)